src/measure_theory/covering/density_theorem.lean

```diff
@@ -0,0 +1,134 @@
+
+/-- A measure `μ` is said to be a doubling measure if there exists a constant `C` such that for
+all sufficiently small radii `ε`, and for any centre, the measure of a ball of radius `2 * ε` is
+bounded by `C` times the measure of the concentric ball of radius `ε`. -/
+class is_doubling_measure {α : Type*} [metric_space α] [measurable_space α] (μ : measure α) :=
+(exists_measure_closed_ball_le_mul [] :
+  ∃ (C : ℝ≥0), ∀ᶠ ε in 𝓝[>] 0, ∀ x, μ (closed_ball x (2 * ε)) ≤ C * μ (closed_ball x ε))
+
+namespace is_doubling_measure
+
+variables {α : Type*} [metric_space α] [measurable_space α] (μ : measure α) [is_doubling_measure μ]
+
+/-- The doubling constant of a doubling measure. -/
+def doubling_constant : ℝ≥0 := classical.some $ exists_measure_closed_ball_le_mul μ
+
+lemma exists_measure_closed_ball_le_mul' :
+  ∀ᶠ ε in 𝓝[>] 0, ∀ x, μ (closed_ball x (2 * ε)) ≤ (doubling_constant μ) * μ (closed_ball x ε) :=
+classical.some_spec $ exists_measure_closed_ball_le_mul μ
+
+lemma exists_measure_closed_ball_le_mul_of_mem_Ioc (t : ℝ) (ht : t ∈ Ioc (0 : ℝ) 8) :
+  ∀ᶠ ε in 𝓝[>] 0, ∀ x,
+    μ (closed_ball x (t * ε)) ≤ ↑((doubling_constant μ)^3) * μ (closed_ball x ε) :=
+begin
+  let C := doubling_constant μ,
+  let h := exists_measure_closed_ball_le_mul' μ,
+  let hp : 0 < (2 : ℝ) := two_pos,
+  apply ((eventually_nhds_within_pos_mul_left hp $ eventually_nhds_within_pos_mul_left hp h).and $
+    (eventually_nhds_within_pos_mul_left hp h).and h).mono,
+  rintros ε ⟨h₁, h₂, h₃⟩ x,
+  calc μ (closed_ball x (t * ε))
+        ≤ μ (closed_ball x (8 * ε)) : measure_mono _
+    ... = μ (closed_ball x (2 * (2 * (2 * ε)))) : by { simp only [← mul_assoc], norm_num, }
+    ... ≤ C * μ (closed_ball x (2 * (2 * ε))) : h₁ x
+    ... ≤ C * (C * μ (closed_ball x (2 * ε))) : ennreal.mul_left_mono $ h₂ x
+    ... ≤ C * (C * (C*μ (closed_ball x ε))) : ennreal.mul_left_mono $ ennreal.mul_left_mono $ h₃ x
+    ... = ↑(C^3) * μ (closed_ball x ε) : by simp only [← mul_assoc, ← pow_three, ennreal.coe_pow],
+  cases le_or_gt 0 ε with hε hε,
+  { exact closed_ball_subset_closed_ball (mul_le_mul_of_nonneg_right ht.2 hε), },
+  { rw [closed_ball_eq_empty.mpr (mul_neg_of_pos_of_neg ht.1 hε),
+        closed_ball_eq_empty.mpr (by linarith : 8 * ε < 0)], },
+end
+
+variables [proper_space α] [borel_space α] [is_locally_finite_measure μ]
+
+/-- The Vitali family of a doubling measure. -/
+def vitali_family : vitali_family μ :=
+vitali.vitali_family μ ((doubling_constant μ)^3) $ λ x ε hε,
+begin
+  have h := forall_eventually_of_eventually_forall
+    (exists_measure_closed_ball_le_mul_of_mem_Ioc μ 6 $ by norm_num),
+  simpa only [exists_prop] using ((eventually_nhds_within_pos_mem_Ioc hε).and (h x)).exists,
+end
+
+/-- A version of *Lebesgue's density theorem* for a sequence of closed balls whose centres are
+not required to be fixed.
+
+See also `besicovitch.ae_tendsto_measure_inter_div`. -/
+lemma ae_tendsto_measure_inter_div (S : set α) (K : ℝ) (hK : K ∈ unit_interval) :
```

Comment:
You don't need `hK`: the theorem is true for any `K` (but maybe you will need to build some more API to prove this, and in particular to have a family of vitali families, not just one).

Reply:
I was also being lazy here. I'll come back to this with the generalisation.

Reply:
On second thought, it does not seem to be easy to remove the hypothesis `hK`. Am I missing the obvious? Is it possible to defer turn this work into a TODO?